src/main/java/shop/service/action/DecreaseActionHandler.java

```diff
@@ -0,0 +1,25 @@
+package shop.service.action;
+
+import shop.dao.FruitDaoImpl;
+import shop.model.Fruit;
+
+public class DecreaseActionHandler implements ActionHandler {
+    private final FruitDaoImpl fruitDao;
+
+    public DecreaseActionHandler() {
+        fruitDao = new FruitDaoImpl();
+    }
```

Comment:
you should not depend on the implementation
please pass the dependency through the constructor